gstreamer-sdp/src/s_d_p_media.rs

```diff
@@ -28,8 +28,14 @@ glib_wrapper! {
     pub struct SDPMedia(Boxed<ffi::GstSDPMedia>);
 
     match fn {
-        copy => |ptr| ffi::gst_sdp_media_copy(mut_override(ptr)),
-        free => |ptr| ffi::gst_sdp_media_free(ptr),
+        copy => |ptr| {
+            let mut copy = glib_ffi::g_malloc(mem::size_of::<ffi::GstSDPMedia>()) as *mut ffi::GstSDPMedia;
+            ffi::gst_sdp_media_copy(ptr as *const ffi::GstSDPMedia, &mut copy);
+            copy
```

Comment:
This is actually wrong, sorry.
```rust
            let mut copy = ptr::null_mut();
            ffi::gst_sdp_media_copy(ptr as *const ffi::GstSDPMedia, &mut copy);
            copy
```

Reply:
Curious, when is it idiomatic to use the following different methods for passing memory to a `ffi` "constructor" method. After observing the usage of each I haven't made any sensible conclusions so far... XD

  - `mem::uninitialized()`
  - `mem::zeroed()` (It looks like this method is always preferred over `mem::uninitialized`? It's possible for uninitialized memory to contain garbage?)
  - `ptr::null_mut()`
  - `g_malloc0` (After reading docs [here](https://developer.gnome.org/glib/stable/glib-Memory-Allocation.html#g-malloc) I can see why this is preferred)

Reply:
`uninitialized`/`g_malloc` are fine if the memory is going to be fully initialized. `zeroed`/`g_malloc0` otherwise. The `mem::` functions for stack allocations, `g_malloc` for heap allocation.

Reply:
Oh and I don't really trust the SDP library to properly initialize everything. It seems to require you to zero-initialize things. As said, uncommon and annoying API :)

Reply:
Hmmmm, what is the rationale for using `ptr::null_mut()` here versus `g_malloc`. Both methods are allocating memory on the heap?

Reply:
First is on the stack, second on the heap. The first here because the function will allocate for us and we only need to provide a pointer where it can write to